### PR TITLE
New module: cnos_portchannel

### PR DESCRIPTION
**Pull request by dkasberg**
##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

cnos_portchannel module

##### ANSIBLE VERSION

```
ansible 2.3.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### SUMMARY

```

```

### CODE REVIEW

File: lib/ansible/modules/network/cnos/cnos_portchannel.py [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+        or vid-management or vlan-name, counter for load interval, policy input name, all or Copp class name to attach,
+        qos, queing, Enter the allowed traffic level, ipv6]
+    interfaceArg4:
+        description:
+            - This is an overloaded Port Channel fourth argument. Usage of this argument can be found is the User Guide referenced above.
+        required: No
+        default: Null
+        choices: [key-chain, key-id, keyed-md5 or keyed-sha1 or meticulous-keyed-md5 or meticulous-keyed-sha1 or simple, Interval value in milliseconds,
+         Destination IP (Both IPV4 and IPV6),in or out, MAC address, Time-out value in seconds, class-id, request, Specify the IPv4 address,
+         OSPF area ID as a decimal value, OSPF area ID in IP address format, anycast or secondary, ethernet, vlan, MAC (hardware) address in HHHH.HHHH.HHHH format,
```
Comment (mattclay) [inline lib/ansible/modules/network/cnos/cnos_portchannel.py]: CI failure due to PEP 8 issue:

```
2017-02-17 18:00:17 ERROR: PEP 8: lib/ansible/modules/network/cnos/cnos_portchannel.py:90:161: E501 line too long (163 > 160 characters) (current)
```

File: lib/ansible/modules/network/cnos/cnos_portchannel.py [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+        or vid-management or vlan-name, counter for load interval, policy input name, all or Copp class name to attach,
+        qos, queing, Enter the allowed traffic level, ipv6]
+    interfaceArg4:
+        description:
+            - This is an overloaded Port Channel fourth argument. Usage of this argument can be found is the User Guide referenced above.
+        required: No
+        default: Null
+        choices: [key-chain, key-id, keyed-md5 or keyed-sha1 or meticulous-keyed-md5 or meticulous-keyed-sha1 or simple, Interval value in milliseconds,
+         Destination IP (Both IPV4 and IPV6),in or out, MAC address, Time-out value in seconds, class-id, request, Specify the IPv4 address,
+         OSPF area ID as a decimal value, OSPF area ID in IP address format, anycast or secondary, ethernet, vlan,
```
Comment (mattclay) [inline lib/ansible/modules/network/cnos/cnos_portchannel.py]: CI failure due to PEP 8 issue:

```
2017-02-21 22:09:43 ERROR: PEP 8: lib/ansible/modules/network/cnos/cnos_portchannel.py:90:115: W291 trailing whitespace (current)
```

You can run the PEP 8 tests locally with `make pep8`.

**Comment by mattclay** [comment]
CI failure due to yamllint error:

```
2017-02-21 23:31:50 Run command: yamllint test/integration/targets/cnos_portchannel/tasks/main.yml test/integration/targets/cnos_portchannel/vars/main.yml
2017-02-21 23:31:50 test/integration/targets/cnos_portchannel/vars/main.yml
2017-02-21 23:31:50   19:101    error    too few spaces after comma  (commas)
```

You can test this locally with: `yamllint test/integration/targets/cnos_portchannel/tasks/main.yml test/integration/targets/cnos_portchannel/vars/main.yml`

**Comment by mattclay** [comment]
Tests for `cnos` modules should go into `test/integration/roles/` instead of `test/integration/targets/`, since we're not able to run them as part of CI or through `ansible-test` as-is. Previously, the suggestion was to create an empty `aliases` file for tests in `test/integration/targets/`, but this is no longer recommended. In the future, if a virtual appliance, such as an AWS AMI image, becomes available, we can revisit the possibility of running `cnos` module tests via `ansible-test`.

needs_revision

**Review by gundalow — APPROVED** [review]

**Pull request merged**